### PR TITLE
Remove Parliament JSON

### PR DESCRIPTION
**Pull request by 31453**

### CODE REVIEW

File: parliament/parliament.js [hunk 1]
```diff
@@ -202,16 +198,195 @@ function newError (code, msg) {
...
+   * @returns {Parliament} parliament - The requested parliament
+   */
+  static async apiGetParliament (req, res) {
+    try {
+      const { body: { _source: parliament } } = await Parliament.getParliament();
+
+      Parliament.#cache.set('parliament', parliament);
+
+      if (!req.user.hasRole('parliamentAdmin')) {
+        delete parliament.settings;
```
Comment (31453) [inline parliament/parliament.js:330]: copy this before deleting otherwise it deletes the cache

File: parliament/parliament.js [hunk 2]
```diff
@@ -202,16 +198,195 @@ function newError (code, msg) {
...
+   *
+   * Updates a parliament by id (name).
+   * @name /parliament
+   * @returns {boolean} success - Whether the operation was successful.
+   * @returns {string} text - The success/error message to (optionally) display to the user.
+   * @returns {Parliament} parliament - The updated parliament.
+   */
+  static async apiUpdateParliament (req, res) {
+    try {
+      await Parliament.setParliament(req.body);
```
Comment (31453) [inline parliament/parliament.js:353]: do a get before a set (for settings) and maybe validate the parliament

File: parliament/parliament.js [hunk 3]
```diff
@@ -202,16 +198,195 @@ function newError (code, msg) {
...
+      return res.json({ success: true, text: 'Parliament updated successfully', parliament: req.body });
```
Comment (31453) [inline parliament/parliament.js:354]: don't just send this back. If I need this on the client, use what I fetched from above

File: parliament/parliament.js [hunk 4]
```diff
@@ -202,16 +198,195 @@ function newError (code, msg) {
...
+    let parliament = Parliament.#cache.get('parliament');
+
+    if (!parliament) {
+      const { body: { _source: updatedParliament } } = await Parliament.getParliament();
+      Parliament.#cache.set('parliament', updatedParliament);
+      parliament = updatedParliament;
+    }
+
+    let val = Parliament.settingsDefault.general[type];
+    if (parliament?.settings?.general[type]) {
```
Comment (31453) [inline parliament/parliament.js:379]: return parliament?.settings?.general[type] ?? Parliament.settingsDefault.general[type];

File: parliament/parliament.js [hunk 5]
```diff
@@ -370,6 +548,158 @@ function findIssue (clusterId, issueType, node) {
...
+      throw new Error(e);
+    }
+
+    // notify of upgrade success
+    console.log(`SUCCESS - Parliament upgraded to version ${MIN_PARLIAMENT_VERSION}`);
+  }
+
+  // create parliament in db if it doesn't exist
+  // (there was no parliament file file to upgrade from)
+  try {
```
Comment (31453) [inline parliament/parliament.js:605]: put this above `parliamentFile = ` and don't load parliamentFile or check for upgrade if it exists in db

File: parliament/parliament.js [hunk 6]
```diff
@@ -505,209 +835,111 @@ function getStats (cluster) {
...
+    });
+  });
 }
 
-// Chains all promises for requests for health and stats to update each cluster
-// in the parliament
-function updateParliament () {
+// Chains all promises for requests for health and stats for each cluster in the parliament
+// this also sets all the issues in the issues.json file
+async function updateParliament () {
```
Comment (31453) [inline parliament/parliament.js:940]: Take it out of the class and use it directly in the endpoint definition

File: parliament/parliament.js [hunk 7]
```diff
@@ -1121,40 +1169,56 @@ app.post('/parliament/api/groups', [isAdmin, checkCookieToken], (req, res, next)
...
-      foundGroup = true;
-      break;
+app.delete('/parliament/api/groups/:id', [isAdmin, checkCookieToken], async (req, res, next) => {
+  try {
+    const { body: { _source: parliament } } = await Parliament.getParliament();
+
+    let index = 0;
+    let foundGroup = false;
+    for (const group of parliament.groups) {
+      console.log(group.id, req.params.id);
```
Comment (31453) [inline parliament/parliament.js:1197]: remove this!

File: parliament/upgrade.js [hunk 8]
```diff
@@ -7,16 +7,23 @@
...
 /**
  * Upgrades the parliament object to the latest version
  * @param {object} parliament the parliament object to upgrade
+ * @param {object} ArkimeConfig the ArkimeConfig object
+ * @param {string} parliamentName the name of the parliament (must be unique)
+ * @param {object} issues the issues object to update
+ * @param {object} Parliament the Parliament class
  */
-exports.upgrade = async function (parliament, ArkimeConfig) {
+exports.upgrade = async function (parliament, ArkimeConfig, parliamentName, issues, Parliament) {
```
Comment (31453) [inline parliament/upgrade.js:26]: maybe don't pass in ArkimeConfig - there was an issue with it not being initialized properly, change the order of the import
parliamentName can come from Parliament.#name (make it accessible)
can read issues directly from json
Parliament and be imported from parliament.js?

File: tests/parliament.t [hunk 9]
```diff
@@ -224,62 +218,69 @@ eq_or_diff($result, from_json('{"success":false,"text":"A group must have a stri
...
+# Restore defaults
+$result = parliamentPutToken("/parliament/api/settings/restoreDefaults?molochRegressionUser=parliamentAdminP", '{}', $parliamentAdminToken);
+delete $result->{settings};
+eq_or_diff($result, from_json('{"success":true, "text":"Successfully restored default settings."}'));
 
 # Get parliament
 $result = parliamentGetToken("/parliament/api/parliament?molochRegressionUser=parliamentAdminP", $parliamentAdminToken);
-eq_or_diff($result, from_json('{"authMode": "digest", "groups": [{"clusters": [], "id": 0, "title": "the title"}, {"clusters": [], "description": "UP description for 2", "id": 1, "title": "UP the second title"}], "version": ' . $version .'}'));
+delete $result->{settings};
+eq_or_diff($result, from_json('{"groups": [{"clusters": [], "id": "' . $firstGroupId . '", "title": "the title"}, {"clusters": [], "description": "UP description for 2", "id":  "' . $secondGroupId . '", "title": "UP the second title"}], "name": "Parliament"}'));
```
Comment (31453) [inline tests/parliament.t:254]: qq!

File: db/db.pl [hunk 10]
```diff
@@ -5683,6 +5683,101 @@ sub notifiersAddMissingProps
...
+  parliamentUpdate();
+}
+
+sub parliamentUpdate
+{
+    my $mapping = '
+{
+  "_source": {"enabled": "true"},
+  "dynamic": "strict",
+  "properties": {
```
Comment (31453) [inline db/db.pl:5710]: add
```
"dynamic_templates": [
    {
      "string_template": {
        "match_mapping_type": "string",
        "mapping": {
          "type": "keyword"
        }
      }
    }
  ]
```

Comment (awick) [reply db/db.pl:5711]: it looks like this got added to views and not parliament?

Comment (31453) [reply db/db.pl:5711]: 🤦

File: parliament/parliament.js [hunk 11]
```diff
@@ -202,16 +198,588 @@ function newError (code, msg) {
...
+  }
+
+  static async createParliament (parliament) {
+    return Parliament.#esclient.create({
+      index: Parliament.#parliamentIndex, body: parliament, id: Parliament.name, timeout: '10m'
+    });
+  }
+
+  static async setParliament (parliament) {
+    return Parliament.#esclient.index({
```
Comment (awick) [inline parliament/parliament.js:255]: this should either delete the cache or replace the cache

File: parliament/parliament.js [hunk 12]
```diff
@@ -202,16 +198,588 @@ function newError (code, msg) {
...
+      for (const s in req.body.settings.general) {
+        let setting = req.body.settings.general[s];
+
+        if (s !== 'hostname' && s !== 'includeUrl') {
+          if (isNaN(setting)) {
+            return res.serverError(422, `${s} must be a number.`);
+          } else {
+            setting = parseInt(setting);
+          }
+        }
```
Comment (awick) [inline parliament/parliament.js:364]: should we verify hostname/includeURL are ArkimeUtil.isString?

File: parliament/parliament.js [hunk 13]
```diff
@@ -202,16 +198,588 @@ function newError (code, msg) {
...
+   * @name /parliament
+   * @returns {boolean} success - Whether the operation was successful.
+   * @returns {string} text - The success/error message to (optionally) display to the user.
+   * @returns {Parliament} parliament - The updated parliament.
+   */
+  static async apiUpdateParliamentOrder (req, res) {
+    try {
+      const { body: { _source: parliament } } = await Parliament.getParliament();
+
+      if (!req.body.oldIdx && req.body.newIdx) {
```
Comment (awick) [inline parliament/parliament.js:417]: ArkimeUtil.isString instead?

File: parliament/parliament.js [hunk 14]
```diff
@@ -202,16 +198,588 @@ function newError (code, msg) {
...
+        return res.serverError(500, 'Error updating parliament order');
+      }
+
+      if (req.body.newGroupId) { // we're rearranging clusters
```
Comment (awick) [inline parliament/parliament.js:421]: same, in general trying to banish all if (req and if (!req from code

File: parliament/parliament.js (hunk 13 above)
Comment (31453) [reply parliament/parliament.js:417]: instead of what?

Comment (awick) [reply parliament/parliament.js:417]: just checking for existance

**Pull request merged**